churaverse-plugins-server/src/churarenPlugin/churarenBossPlugin/churarenBossPlugin.ts

```diff
@@ -0,0 +1,236 @@
+      const dest = boss.position.copy()
+      dest.gridX = position.gridX + direction.x * bossSpeedMultiplier
+      dest.gridY = position.gridY + direction.y * bossSpeedMultiplier
+      if (this.isBossWalkInMap(dest, currentMap)) {
+        const velocity = { x: direction.x * speed * bossSpeedMultiplier, y: direction.y * speed * bossSpeedMultiplier }
+        this.networkPluginStore.messageSender.send(
+          new BossWalkMessage({
+            startPos: position.toVector() as Vector & SendableObject,
+            dest: dest.toVector() as Vector & SendableObject,
+            direction,
+            speed,
+            bossId,
+          })
+        )
+        boss.walk(position, direction, velocity)
+        return
+      }
+    }
+  }
+
+  private onCollisionPlayer(boss: Boss, player: Player): void {
+    if (player.isDead) return
+    if (!boss.isCollidable) return
+    if (this.churarenGameInfo === undefined || !this.churarenGameInfo.participantIds.includes(player.id)) return
+    const collisionBossDamageCause = new CollisionBossDamageCause(boss)
+    const livingDamageEvent = new LivingDamageEvent(player, collisionBossDamageCause, boss.power)
+    this.bus.post(livingDamageEvent)
+  }
+
+  private readonly onLivingDamage = (ev: LivingDamageEvent): void => {
+    if (!(ev.target instanceof Boss)) return
+    const boss = this.bossPluginStore.bosses.get(ev.target.bossId)
+    if (boss === undefined) return
+
+    if (ev.cause instanceof ChurarenWeaponDamageCause) {
+      const weaponDamageMessage = new WeaponDamageMessage({
+        targetId: boss.bossId,
+        cause: ev.cause.churarenWeaponName,
+        weaponId: ev.cause.churarenWeapon.id,
+        amount: ev.amount,
+      })
+      this.networkPluginStore.messageSender.send(weaponDamageMessage)
+      boss.damage(ev.amount)
+    }
+
+    if (boss.isDead) {
+      const bossDespawnMessage = new BossDespawnMessage({ bossId: boss.bossId })
+      this.bossPluginStore.bosses.delete(boss.bossId)
+      this.networkPluginStore.messageSender.send(bossDespawnMessage)
+
+      const updateChurarenUi = new ChurarenResultEvent('win')
+      this.bus.post(updateChurarenUi)
+    }
+  }
+
+  private readonly onBossDespawn = (): void => {
+    this.bossPluginStore.bosses.getAllId().forEach((bossId) => {
+      const boss = this.bossPluginStore.bosses.get(bossId)
+      if (boss === undefined) return
+      const bossDespawnMessage = new BossDespawnMessage({ bossId: boss.bossId })
+      this.bossPluginStore.bosses.delete(boss.bossId)
+      this.networkPluginStore.messageSender.send(bossDespawnMessage)
+    })
+  }
+
+  private readonly registerOnOverlap = (ev: RegisterOnOverlapEvent): void => {
+    ev.collisionDetector.register(
+      this.bossPluginStore.bosses,
+      this.store.of('playerPlugin').players,
+      this.onCollisionPlayer.bind(this)
+    )
+  }
+
+  private isBossWalkInMap(dest: Position, currentMap: WorldMap): boolean {
+    return (
+      dest.x > this.halfBossSize &&
```

Comment:
この関数でしか使っていないプロパティで、かつ`CHURAREN_BOSS_SIZE`に完全に依存していて記述が今後も変わることが考えられないので、この関数内のローカルな変数として宣言してしまう方がいいかなと思います。
```suggestion
    const halfBossSize = CHURAREN_BOSS_SIZE / 2
    return (
      dest.x > this.halfBossSize &&
```